### PR TITLE
Add support for ActionScript 3.0 & MXML

### PR DESCRIPTION
**Pull request by jcs090218**

### CODE REVIEW

**Review by yyoncho — APPROVED** [review]
Looks good!

File: clients/lsp-actionscript.el [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+ 'as3mxml
+ '(:system "as3mxml")
+ `(:download :url lsp-actionscript-server-download-url
+             :store-path ,(lsp-actionscript--extension-path-zip)))
+
+(lsp-register-client
+ (make-lsp-client
+  :new-connection (lsp-stdio-connection
+                   #'lsp-actionscript--server-command
+                   (lambda () (f-exists? (lsp-actionscript--extension-path-zip))))
```
[the remark this answers is not in the document]
Comment (jcs090218) [reply clients/lsp-actionscript.el:114]: Any suggestion avoiding this problem? 😕

[the remark this answers is not in the document]
Comment (yyoncho) [reply clients/lsp-actionscript.el:114]: you should check if something from the unzipped content is present.

File: clients/lsp-actionscript.el [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  "Return full path of the downloaded extension."
+  (f-join lsp-actionscript-server-store-path lsp-actionscript-extension-name))
+
+(defun lsp-actionscript--extension-dir ()
+  "Return as3mxml extension path."
+  (f-join (lsp-actionscript--extension-root) "extension"))
+
+(defun lsp-actionscript--server-command ()
+  "Startup command for ActionScript language server."
+  (list "java"
```
[the remark this answers is not in the document]
Comment (jcs090218) [reply clients/lsp-actionscript.el]: Done. I have extracted java to a variable. ;)

**Comment by jcs090218** [comment]
I will merge this first then make error handling (unzipping) on the next patch. ;)

**Pull request merged**